Backend/src/middleware/auth.middlewares.ts

```diff
@@ -0,0 +1,48 @@
+import { Request, Response, NextFunction } from "express";
+import jwt from "jsonwebtoken";
+import {
+  AUTH_MESSAGES,
+  AUTH_HEADERS,
+  AUTH_PREFIXES,
+  AUTH_DEFAULTS,
+} from "../constants/MIDDLEWARE/middlewareConstants";
+
+// Extend the Request type to include user
+declare global {
+  namespace Express {
+    interface Request {
+      user?: { id: number };
+    }
+  }
+}
+
+export class AuthMiddleware {
+  static async authenticateUser(
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) {
+    const authHeader = req.headers[AUTH_HEADERS.AUTHORIZATION];
+
+    // Ensure authHeader is a string before calling startsWith
+    const token =
+      typeof authHeader === "string" &&
+      authHeader.startsWith(AUTH_PREFIXES.BEARER)
+        ? authHeader.split(" ")[1]
+        : null;
+
+    if (!token)
+      return res.status(401).json({ message: AUTH_MESSAGES.ACCESS_DENIED });
```

Comment:
Instead of **Access Denied** we should use **401 Unauthorized**. As this will be checking if user is authorize as you have mentioned it in line no. 25.

Reply:
Changes the error message from Access Denied to Unauthorized